test/types.array.test.js

```diff
@@ -1899,4 +1899,18 @@ describe('types array', function() {
 
     m.Schema.Types.Array.options.castNonArrays = true;
   });
+
+  it('supports setting nested arrays directly (gh-13372)', async function() {
```

Comment:
Maybe have "duplicate" test where we check that 0 is properly handled, as expected. Because this is why you do the != null check in array.js